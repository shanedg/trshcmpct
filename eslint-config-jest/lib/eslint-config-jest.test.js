const eslint = require('eslint');

const ESLint = eslint.ESLint;

describe('eslint-config', () => {
  let eslintApi;

  beforeAll(() => {
    eslintApi = new ESLint({
      overrideConfigFile: 'lib/eslint-config-jest.js',
      useEslintrc: false,
    });
  });

  describe('calculated config snapshots', () => {
    let config;

    beforeAll(async () => {
      // This file doesn't actually have to exist and the path doesn't matter
      // since we override the config file and useEslintrc is false.
      config = await eslintApi.calculateConfigForFile('anyFile.js');
    });

    it('env', () => {
      expect(config.env).toMatchSnapshot();
    });

    it('rules', () => {
      expect(config.rules).toMatchSnapshot();
    });
  });

});
