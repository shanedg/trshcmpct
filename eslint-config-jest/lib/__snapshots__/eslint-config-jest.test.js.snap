// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eslint-config-jest calculated config snapshots env 1`] = `
{
  "jest/globals": true,
}
`;

exports[`eslint-config-jest calculated config snapshots rules 1`] = `
{
  "jest/consistent-test-it": [
    "error",
  ],
  "jest/expect-expect": [
    "warn",
  ],
  "jest/no-alias-methods": [
    "error",
  ],
  "jest/no-commented-out-tests": [
    "warn",
  ],
  "jest/no-conditional-expect": [
    "error",
  ],
  "jest/no-deprecated-functions": [
    "error",
  ],
  "jest/no-disabled-tests": [
    "warn",
  ],
  "jest/no-done-callback": [
    "error",
  ],
  "jest/no-export": [
    "error",
  ],
  "jest/no-focused-tests": [
    "error",
  ],
  "jest/no-identical-title": [
    "error",
  ],
  "jest/no-interpolation-in-snapshots": [
    "error",
  ],
  "jest/no-jasmine-globals": [
    "error",
  ],
  "jest/no-mocks-import": [
    "error",
  ],
  "jest/no-standalone-expect": [
    "error",
  ],
  "jest/no-test-prefixes": [
    "error",
  ],
  "jest/valid-describe-callback": [
    "error",
  ],
  "jest/valid-expect": [
    "error",
  ],
  "jest/valid-expect-in-promise": [
    "error",
  ],
  "jest/valid-title": [
    "error",
  ],
  "quotes": [
    "warn",
    "single",
    {
      "allowTemplateLiterals": true,
      "avoidEscape": true,
    },
  ],
}
`;
